### ISSUE TITLE
Some clarifications regarding results on different datasets

### ISSUE THREAD

**Issue by kirk86** [satwik77/Transformer-Simplicity#1]
Hi there, I've tried to replicate some of the results for some datasets but couldn't get good results.
For instance on the dataset `sparity40_5k` (also tried `sparity40_25h` and `sparse_parity4a`) as you can see below training for 1500 epochs doesn't get beyond 52% validation accuracy.

Do you have any suggestions?
On a side note I've noticed that for most datasets the validation dataset has way more samples than the training dataset, why is that? Shouldn't the training set have more samples and the validation set a smaller fraction?

For instance, the  `sparity40_5k` has 5k samples in the `train.pkl` and 10k in the `dev.pkl`,  similar for `sparity40_25h` with 2.5k samples in `train.pkl` and 10k in `dev.pkl`

```
{
    "trafo_sparity_40_5k": {
        "run_name": "trafo_sparity_40_5k",
        "val_score": 0.5214000000000001,
        "train_acc": 0.9912000000000001,
        "best_epoch": 929,
        "dataset": "sparity40_5k",
        "heads": 4,
        "d_model": 64,
        "depth": 4,
        "dropout": 0.1,
        "lr": 0.001,
        "batch_size": 500,
        "epochs": 1500,
        "opt": "adam"
    }
}
```
